src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
             {
                 IFormCollection form = await Request.ReadFormAsync();
                 Request.Body.Seek(0, SeekOrigin.Begin);
-
-                openIdConnectMessage = new OpenIdConnectMessage(form);
+                message = new OpenIdConnectMessage(form);
             }
 
-            if (openIdConnectMessage == null)
+            if (message == null)
```

Comment:
An advantage? Well, only one: it's supported everywhere :smile: 

That would be great if the experts you mentioned could share more information about the weaknesses `response_mode=query` has/may have. Specially since it's heavily used by the OAuth2 (Google/Facebook) and OAuth1 social providers.